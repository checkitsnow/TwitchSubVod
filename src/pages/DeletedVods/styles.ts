import styled, { keyframes } from 'styled-components';
import { appearFromTop } from '../../components/Footer/styles';

const appearFromBottom = keyframes`
  from {
    opacity: 0;
    transform: translateY(2rem);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;

export const spin = keyframes`
  100% {
    transform: rotate(360deg)
  }
`;

export const Container = styled.div`
  min-height: 100vh;
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 0 2rem;
`;

export const AnimationContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  animation: ${appearFromBottom} 0.5s;

  form {
    display: flex;
    justify-content: center;
    flex-direction: column;
  }
  .input-box {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    background: var(--light-background);
    height: 2rem;
    max-width: 20rem;
    padding: 0.5rem;
    margin-bottom: 0.5rem;

    input {
      flex: 1;
      background: var(--light-background);
      color: var(--text);
      text-align: center;
      appearance: textfield;
    }
  }

  .quality-selection {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 0.5rem;

    select {
      width: 50%;
      background-color: var(--light-background);
      color: var(--text);
      border: none;
      padding: 0.25rem;
    }
  }

  button[type='submit'] {
    display: flex;
    align-items: center;
    justify-content: center;
    background: var(--purple);
    height: 2rem;
    width: 20rem;
    border: 2px solid var(--purple);
    color: var(--text);

    & svg {
      margin-right: 0.5rem;
    }
  }

  .video-container {
    max-width: 60rem;
    margin: 2rem 0;

    animation: ${appearFromTop} 0.5s ease-out;
  }

  span {
    margin-top: 2rem;

    & svg {
      animation: ${spin} 4s linear infinite;
    }
  }
`;
