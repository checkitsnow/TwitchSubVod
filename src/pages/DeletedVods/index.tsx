import React, { useEffect, useState } from 'react';
import ReactGA from 'react-ga';
import axios from 'axios';
import ReactPlayer from 'react-player';

import { FiSearch, FiLoader } from 'react-icons/fi';

import { Container, AnimationContainer } from './styles';
import LinkBox from '../../components/LinkBox';
import InfoModal from '../../components/InfoModal';
import Footer from '../../components/Footer';

const DeletedVods: React.FC = () => {
  useEffect(() => {
    ReactGA.initialize(`${process.env.REACT_APP_GOOGLE_TRACKING}`, {
      testMode: process.env.NODE_ENV === 'test',
    });
    ReactGA.pageview('/DeletedVods');
  }, []);

  const [vodId, setVodId] = useState('');
  const [data, setData] = useState('');
  const [vodQuality, setVodQuality] = useState('chunked');
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  const handleSubmit = async () => {
    if (vodId.length === 11) {
      try {
        setLoading(true);
        const response = await axios.get(
          `${process.env.REACT_APP_CORS}${process.env.REACT_APP_DELETED_VOD_WORKER}${vodId}`,
        );

        const link = response.data
          .split(`https://vod-metro.twitch.tv/`)[1]
          .split('/')[0];

        let finalLink = `${process.env.REACT_APP_CORS}https://vod-metro.twitch.tv/${link}/${vodQuality}/index-dvr.m3u8`;
        setData(finalLink);
        setLoading(false);

        const responseVodUrl = await axios.get(`${finalLink}`);
        if (responseVodUrl.status === 200 || responseVodUrl.status === 304) {
          setError(false);
        } else {
          setError(true);
        }

        ReactGA.event({
          category: 'Button',
          action: `User searched a deleted vod ${vodId}`,
        });
      } catch (err) {
        console.warn(err);
        setError(true);
        setLoading(false);
      }
    } else {
      alert('Invalid Vod ID');
    }
  };

  return (
    <Container>
      <AnimationContainer>
        <h1>Deleted Twitch Vods</h1>

        <form
          onSubmit={(event) => {
            event.preventDefault();
          }}
        >
          <div className="input-box">
            <input
              type="number"
              name="vodId"
              onChange={(event) => setVodId(event.target.value)}
              value={vodId}
              placeholder="Vod ID"
            />
            <InfoModal />
          </div>

          <div className="quality-selection">
            <label htmlFor="quality">Select a quality: </label>
            <select
              name="quality"
              id="quality"
              onChange={(event) => setVodQuality(event.target.value)}
            >
              <option defaultValue="chunked">Source</option>
              <option value="720p60">720p60</option>
              <option value="720p30">720p30</option>
              <option value="480p30">480p</option>
              <option value="360p30">360p</option>
              <option value="160p30">160p</option>
              <option value="audio_only">audio</option>
            </select>
          </div>

          <button type="submit" onClick={handleSubmit} aria-label="submit">
            <FiSearch size={14} />
            Search
          </button>
        </form>

        <LinkBox home />

        {data && !error ? (
          <>
            <div className="video-container">
              <ReactPlayer
                key={data}
                url={data}
                controls
                width="100%"
                height="100%"
              />
            </div>
          </>
        ) : loading ? (
          <span>
            <FiLoader size={32} />
          </span>
        ) : (
          error && (
            <span>The VOD you're looking for is no longer available</span>
          )
        )}
      </AnimationContainer>
      <Footer />
    </Container>
  );
};

export default DeletedVods;
