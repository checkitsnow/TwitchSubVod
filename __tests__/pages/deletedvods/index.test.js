import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DeletedVods from '@/pages/deletedvods';

afterEach(cleanup);

test('should render Deleted Vods correctly', async () => {
  const { container } = render(<DeletedVods />);
  expect(container).toMatchSnapshot();
});
