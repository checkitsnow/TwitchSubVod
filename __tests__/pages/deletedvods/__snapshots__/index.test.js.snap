// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Deleted Vods correctly 1`] = `
<div>
  <div
    class="DeletedVods__Container-lbpcr0-0 dNtvmi"
  >
    <div
      class="DeletedVods__AnimationContainer-lbpcr0-1 gNFmSd"
    >
      <h1>
        Deleted Twitch Vods
      </h1>
      <form>
        <div
          class="input-box"
        >
          <input
            name="vodId"
            placeholder="Vod ID"
            type="number"
            value=""
          />
          <div
            class="styles__Container-pgyzn6-0 kCeIyN"
          >
            <div
              class="info-box"
            >
              <p>
                You can get the Vod Id from
                <a
                  href="https://twitchtracker.com"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  twitchtracker.com
                </a>
              </p>
            </div>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="16"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
        <div
          class="styles__Container-ezdgk0-0 dCQtIG"
        >
          <label
            for="quality"
          >
            Select a quality: 
          </label>
          <select
            id="quality"
            name="quality"
          >
            <option>
              Source
            </option>
            <option
              value="720p60"
            >
              720p60
            </option>
            <option
              value="720p30"
            >
              720p30
            </option>
            <option
              value="480p30"
            >
              480p
            </option>
            <option
              value="360p30"
            >
              360p
            </option>
            <option
              value="160p30"
            >
              160p
            </option>
            <option
              value="audio_only"
            >
              audio
            </option>
          </select>
          <div
            class="styles__Container-pgyzn6-0 kCeIyN"
          >
            <div
              class="info-box"
            >
              <p>
                Works best with "Source"
              </p>
            </div>
            <svg
              fill="none"
              height="18"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <line
                x1="12"
                x2="12"
                y1="16"
                y2="12"
              />
              <line
                x1="12"
                x2="12.01"
                y1="8"
                y2="8"
              />
            </svg>
          </div>
        </div>
        <button
          aria-label="submit"
          type="submit"
        >
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Search
        </button>
      </form>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Home
        </span>
      </div>
    </div>
    <div
      class="styles__Container-w3v8va-0 kVLkFL"
    >
      <a
        href="https://ko-fi.com/pogulive"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Buy Me a Coffee at ko-fi.com"
          class="kofi-img"
          src="https://cdn.ko-fi.com/cdn/kofi5.png?v=2"
        />
      </a>
      <a
        href="https://twitter.com/pogulive"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="twitter logo"
          src="https://abs.twimg.com/favicons/twitter.ico"
        />
      </a>
    </div>
  </div>
</div>
`;
